stacktrace.go

```diff
@@ -43,7 +43,7 @@ var (
 	_zapStacktraceVendorContains = addPrefix("/vendor/", _zapStacktracePrefixes...)
 )
 
-func takeStacktrace() string {
+func takeStacktrace(skip *int) string {
```

Comment:
since we default to `0` anyway, can we make this an `int` and pass in `0` if we don't want any skip?